src/main/groovy/edu/oregonstate/mist/idimages/resources/IdImagesResource.groovy

```diff
@@ -35,21 +37,46 @@ class IdImagesResource extends Resource {
     @Produces("image/jpg")
     @Timed
     public Response getByOSUID(@Auth AuthenticatedUser _, @PathParam('id') String id) {
+        Integer bannerPIDM = idImageDAO.getPIDM(id)
+
+        if (!bannerPIDM) {
+            return notFound().type(MediaType.APPLICATION_JSON).build()
+        }
         Blob image = idImageDAO.getByID(id)
 
+        //Return a placeholder image if a person exists but doesn't have an ID image
         if (!image) {
-            return notFound().type(MediaType.APPLICATION_JSON).build()
+            return ok(getImageStream(ImageIO.read(new File("images/defaultImage.jpg")))).build()
         }
+        //Return an ID image of a person
         try {
-            BufferedImage buffImg = ImageIO.read(image.getBinaryStream())
-            ByteArrayOutputStream outputStream = new ByteArrayOutputStream()
-            ImageIO.write(buffImg, "jpg", outputStream)
-            byte[] imageData = outputStream.toByteArray()
-            Response.ok(imageData).build()
+            Response.ok(getImageStream(ImageIO.read(image.getBinaryStream()))).build()
         } catch (Exception e) {
             logger.error("Exception while calling getIDImages", e)
             return internalServerError("Internal server error").
                     type(MediaType.APPLICATION_JSON).build()
         }
     }
+    //Resizes a BufferedImage to a width constraint and calculates the height accordingly
+    public static BufferedImage resizeWidthConstraint(BufferedImage img, int newWidth) {
+        Integer newHeight = ((img.getHeight() * newWidth) / img.getWidth())
+
+        Image tmp = img.getScaledInstance(newWidth, newHeight, Image.SCALE_SMOOTH)
+        BufferedImage resizedImg = new BufferedImage(newWidth, newHeight, img.getType())
+
+        Graphics2D g2d = resizedImg.createGraphics()
+        g2d.drawImage(tmp, 0, 0, null)
+        g2d.dispose()
+
+        resizedImg
+    }
+    //Converts a BufferedImage into a format able to returned by the API
+    public def getImageStream(BufferedImage origImg) {
+        BufferedImage resizedImg = resizeWidthConstraint(origImg, 150)
```

Comment:
Let's use a constant instead of 150. It will make the code more readable and easier to find the value.